docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx

```diff
@@ -457,18 +465,16 @@ class ComponentExample extends React.PureComponent<IComponentExampleProps, IComp
       <div>
         {this.renderApiCodeMenu()}
 
-        {sourceCode != null && (
-          <div>
-            {this.renderCodeEditorMenu()}
-            <Editor
-              setOptions={{ fixedWidthGutter: true, showFoldWidgets: false }}
-              id={`${this.getKebabExamplePath()}-jsx`}
-              value={sourceCode}
-              onChange={this.handleChangeCode}
-              onOutsideClick={this.handleShowCodeInactive}
-            />
-          </div>
-        )}
+        <div>
+          {this.renderCodeEditorMenu()}
+          <Editor
```

Comment:
Should we initialize the sourceCode by default to undefined and test if it is defined then to show the editor? We don't need to render the editor if the sourceCode is not initialized and empty string is not meaning that it is not initialize, it means that there isn't any text inside it. Thoughts?

Reply:
would rather step against initializing `sourceCode` to undefined, as it introduces quite a lot of potential problems where this value is used. If we are declaring possibility for it to be undefined, we should ensure by our logic that this one will be defined at the place where used - and, with that we are starting to rely on runtime assumptions, rather than compile-time contract. This is the main reason for all properties provided in `state` object to be initialized with meaningful defaults.

So, the only question is to address the following point:

>  We don't need to render the editor if the sourceCode is not initialized

Could you, please, suggest what are those cases where we could wrongly initialize it? It seems that we already have a flag that prevents this to happen, few lines above:

```jsx
  if (!showCode) return null
```

Please, let me know if I am missing something

Reply:
Sorry I didn't see the all code, I was just comparing with the previous implementation and it didn't seem logically the same. So, if the showCode is the only source of truth about whether we should show or not show the editor, then yes, this is the way to go.